src/com/gfredericks/schpec/numbers.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.numbers
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(def finite?
+  "Returns true if the given value is an actual finite number"
+  (let [falsies #{Double/POSITIVE_INFINITY
+                  Double/NEGATIVE_INFINITY
+                  Double/NaN}]
+    (fn [x]
+      (and (number? x)
+           (not (contains? falsies x))))))
+
+(s/def ::finite
+  (s/spec finite?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
do you intend this to be just Doubles or other numeric types as well? The generator implies the former while the predicate implies the latter.